Public/Src/Engine/ViewModel/BuildSummary.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BuildXL.ViewModel
+{
+    /// <summary>
+    /// This class collects build information during the build to be displayed in Azure DevOps extensions page as a MarkDown
+    /// </summary>
+    /// <remarks>
+    /// This class is only expected to be instantiated when the users passes /ado on the commandline
+    /// to enable Azure DevOps optimized UI
+    /// </remarks>
+    public class BuildSummary
+    {
+        /// <summary>
+        /// The path where to store the summary
+        /// </summary>
+        private string m_filePath;
+
+        /// <summary>
+        /// This is a model of the tree rendering of the perf regions
+        /// </summary>
+        public PerfTree DurationTree { get; set; }
+
+        /// <nodoc />
+        public CriticalPathSummary CriticalPathSummary { get; } = new CriticalPathSummary();
+
+        /// <nodoc />
+        public CacheSummary CacheSummary { get; } = new CacheSummary();
+
+        /// <nodoc />
+        public List<BuildSummaryPipDiagnostic> PipErrors { get; } = new List<BuildSummaryPipDiagnostic>();
+
+        /// <nodoc />
+        public BuildSummary(string filePath)
+        {
+            m_filePath = filePath;
+        }
+
+        /// <summary>
+        /// This will render the markdown to the predetermined file.
+        /// </summary>
+        public string RenderMarkdown()
+        {
+            using (var writer = new MarkDownWriter(m_filePath))

```

Comment:
>MarkDownWriter [](start = 36, length = 14)

could not find where this ctor is implemented...
depending on the implementation of that ctor, we might need to check that all directories containing the file actually exist before creating the object